target_lik_libraries already used

This error is always there. Any ideas on how to fix it?

-- Linking with these extra flags: -fopenmp -L/gpfs/alpine/proj-shared/csc304/bml/install/lib64/ -lbml_fortran -lbml -L/gpfs/alpine/proj-shared/csc304/magma-2.5.1/lib/ -lmagma  -fopenmp -fopenmp -lpthread -L/sw/summit/essl/6.1.0-2/essl/6.1/lib64 -lesslsmp -L/libmetis
CMake Error at src/CMakeLists.txt:66 (target_link_Libraries):
  The plain signature for target_link_libraries has already been used with
  the target "progress".  All uses of target_link_libraries with a target
  must be either all-keyword or all-plain.

  The uses of the plain signature are here:

   * src/CMakeLists.txt:59 (target_link_libraries)

--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -5,7 +5,7 @@ include_directories(${BML_INCLUDEDIR})
 
 function(progress_example myexample path_to_example)
   add_executable(${myexample} ${path_to_example})
-  target_link_libraries(${myexample}
+  target_link_libraries(${myexample} PUBLIC
     progress
     ${LINK_LIBRARIES})
   set_target_properties(${myexample}

--- a/examples/sp2run/src/CMakeLists.txt
+++ b/examples/sp2run/src/CMakeLists.txt
@@ -18,4 +18,4 @@ set_target_properties(main
   COMPILE_FLAGS ${OpenMP_Fortran_FLAGS}
   LINK_FLAGS ${OpenMP_Fortran_FLAGS})
 
-target_link_libraries(main ${MY_FLAGS})
+target_link_libraries(main PUBLIC ${MY_FLAGS})

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -56,7 +56,7 @@ endif()
 
 if(EXTRA_LINK_FLAGS)
   message(STATUS "Linking with these extra flags: ${EXTRA_LINK_FLAGS}")
-  target_link_libraries(progress ${EXTRA_LINK_FLAGS})
+  target_link_libraries(progress PUBLIC ${EXTRA_LINK_FLAGS})
 endif()
 
 if(TARGET BML::bml)

